demos/coreHTTP/http_demo_s3_download_multithreaded.c

```diff
@@ -0,0 +1,1066 @@
+                   ( int32_t ) xStart,
+                   ( int32_t ) xEnd ) );
+        LogDebug( ( "Request Headers:\n%.*s",
+                    ( int32_t ) xRequestItem.xRequestHeaders.headersLen,
+                    ( char * ) xRequestItem.xRequestHeaders.pBuffer ) );
+
+        xStatus = xQueueSendToBack( xRequestQueue,
+                                    &xRequestItem,
+                                    httpexampleDEMO_TICKS_TO_WAIT );
+
+        /* Ensure request was added to the queue. */
+        if( xStatus == pdFAIL )
+        {
+            LogError( ( "Could not enqueue request." ) );
+        }
+    }
+
+    return xStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool prvNotificationWaitLoop( uint32_t * pulNotification,
+                                     uint32_t ulExpectedBits,
+                                     bool xClearBits )
+{
+    uint32_t ulWaitCounter = 0U;
+    bool ret = true;
+
+    configASSERT( pulNotification != NULL );
+
+    while( ( *pulNotification & ulExpectedBits ) != ulExpectedBits )
+    {
+        xTaskNotifyWait( 0,
+                         ( xClearBits ) ? ulExpectedBits : 0,
+                         pulNotification,
+                         httpexampleDEMO_TICKS_TO_WAIT );
+
+        if( ++ulWaitCounter > httpexampleMAX_WAIT_ITERATIONS )
+        {
+            LogDebug( ( "Loop exceeded maximum wait time. Notification not received.\n" ) );
+            ret = false;
+            break;
+        }
+    }
+
+    return ret;
+}
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvGetS3ObjectFileSize( const HTTPRequestInfo_t * pxRequestInfo,
+                                          QueueHandle_t xRequestQueue,
+                                          QueueHandle_t xResponseQueue,
+                                          size_t * pxFileSize )
+{
+    BaseType_t xStatus = pdPASS;
+    HTTPStatus_t xHTTPStatus = HTTPSuccess;
+
+    /* The location of the file size in pcContentRangeValStr. */
+    char * pcFileSizeStr = NULL;
+
+    /* String to store the Content-Range header value. */
+    char * pcContentRangeValStr = NULL;
+    size_t xContentRangeValStrLength = 0;
+
+    configASSERT( pxRequestInfo != NULL );
+    configASSERT( pxFileSize != NULL );
+
+    LogInfo( ( "Getting file object size from host..." ) );
+
+    /* Request bytes 0 to 0. S3 will respond with a Content-Range
+     * header that contains the size of the file in it. This header will look
+     * like: "Content-Range: bytes 0-0/FILESIZE". The body will have a single
+     * byte that we are ignoring. */
+    xStatus = prvRequestS3ObjectRange( pxRequestInfo,
+                                       xRequestQueue,
+                                       0,
+                                       0 );
+
+    if( xStatus == pdPASS )
+    {
+        xStatus = xQueueReceive( xResponseQueue, &xResponseItem, httpexampleDEMO_TICKS_TO_WAIT );
```

Comment:
Don't we have to check for the errors passed through notification here as well?

Reply:
I've added the check, thanks for catching that.